lightwallet-client-lib/src/main/java/co/electriccoin/lightwallet/client/internal/CoroutineLightWalletClientImpl.kt

```diff
@@ -51,11 +53,23 @@ internal class CoroutineLightWalletClientImpl private constructor(
             .getBlockRange(heightRange.toBlockRange())
     }
 
-    override suspend fun getLatestBlockHeight(): BlockHeightUnsafe {
-        return BlockHeightUnsafe(
-            requireChannel().createStub(singleRequestTimeout)
-                .getLatestBlock(Service.ChainSpec.newBuilder().build()).height
-        )
+    override suspend fun getLatestBlockHeight(): Response<BlockHeightUnsafe> {
+        return try {
+            if (BenchmarkingExt.isBenchmarking()) {
```

Comment:
I might be missing some context so I could be wrong.  If so, dismiss the comment. 

I find this rather odd given the fact that this is an implementation for an abstraction. You probably want a decorator that wraps this method and used specifically as a backend when doing benchmarking instead of having this `if` statement that depends on something that seems to be a global value.

Reply:
We don’t yet have a good way to inject this.  We have a plan to improve this, although it will take multiple iterations to get there.

The refactoring the lightwalletd client will give us an interface that different networking implementations can extend.  Once we have that stabilized (which this PR moves us closer to), then we can pass an alternative networking implementation into the synchronizer which will eliminate the need for this hack.